### PR TITLE
Fixed Target Similar

### LINKED ISSUE #286: Bug: Summon Golem + target similar doesn't target another golem
This should work since summon is supposed to add the summoned unit to the target array

### PR DESCRIPTION
**Pull request by SoulMuncher**
Resolves #286 

Also fixed a similar issue with clone/split and related spells

QA Singleplayer and Load

### CODE REVIEW

File: src/cards/target_similar.ts [hunk 1]
```diff
@@ -58,19 +70,19 @@ export function targetSimilarEffect(numberOfTargets: number) {
...
+        addTarget(newTarget, state);
+      }
     }
-    // Note: New targets must be added AFTER the loop or else it will use the newly added
-    // targets in the loop for this cast of target similar
-    for (let newTarget of newTargets) {
-      addTarget(newTarget, state);
-    }
+
     await animateTargetSimilar(animators);
```
[the remark this answers is not in the document]
Comment (SoulMuncher) [reply src/cards/target_similar.ts:86]: > This PR is reverting a change that I just made this morning to fix another bug, is this intentional?

Moving it outside the loop caused new bugs with clone and similar spells.
Give me a second to double check that the bug you initially fixed is still resolved after this change 👍

**Comment by SoulMuncher** [comment]
Should be ready again. All bugs tested for and no repro 👍

File: src/entity/Unit.ts [hunk 2]
```diff
@@ -159,7 +157,7 @@ export function create(
     }
     const unit: IUnit = Object.assign({
       type: 'unit',
-      id: prediction ? ++lastPredictionUnitId : ++underworld.lastUnitId,
+      id: ++underworld.lastUnitId,
```
Comment (jdoleary) [inline src/entity/Unit.ts:160]: I think this needs to remain different for prediction.  The prediction unit creation runs a LOT (and not the same number of times client to client), we want to make sure the clients' lastUnitId is always the same and so it needs to be different from prediction ids

**Review by jdoleary — CHANGES_REQUESTED** [review]
Looks great!  One small change needed.

**Comment by SoulMuncher** [comment]
Looks good in single and multiplayer @jdoleary 👍

**Review by jdoleary — APPROVED** [review]

**Pull request merged**